### PR TITLE
rate limit: add HTTP local rate limit example with descriptors

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Add an example of server-side rate limiting based on combination of:
- source attribute: app label, expressed as `filter_state['wasm.downstream_peer'].labels['app'].value`
- request header: `user-id` header

### CODE REVIEW

**Comment by kyessenov** [comment]
I'd like to cherry pick this to 1.9. I think we're missing one PR from envoy.

File: test/envoye2e/ratelimit/ratelimit_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"testing"
+	"time"
+
+	"istio.io/proxy/test/envoye2e"
+	"istio.io/proxy/test/envoye2e/driver"
+)
+
+// TestHTTPLocalRatelimit validates that envoy can rate limit based on:
+// - source attribute
```
Comment (bianpengyuan) [inline test/envoye2e/ratelimit/ratelimit_test.go:26]: Suggest adding `extracted by metadata exchange filter` here

**Review by bianpengyuan — COMMENTED** [review]
lgtm

File: test/envoye2e/ratelimit/ratelimit_test.go (hunk 1 above)
Comment (kyessenov) [reply test/envoye2e/ratelimit/ratelimit_test.go]: Done.

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**